instrumentation/rails/lib/opentelemetry/instrumentation/rails/span_subscriber.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      class SpanSubscriber
+        def start(name, id, payload)
+          prev_ctx = OpenTelemetry::Context.current
+
+          span_name = name.split('.')[0..1].reverse.join(' ')
+          span = otel_tracer.start_span(span_name, kind: :internal)
+          OpenTelemetry::Context.current = OpenTelemetry::Trace.context_with_span(span)
```

Comment:
😬 so... technically, `OpenTelemetry::Context.current=` should not be exposed as part of the public API (it should be `@api private`). The `attach` and `detach` methods should really be class methods (see ["Optional Global operations](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/context.md#optional-global-operations)) and they should be used here instead (they should *not* be `@api private`). The main reason to use `attach` and `detach` is that `attach` will return the previous context and `detach` will check that the correct context is being restored.

We need to refactor `Context` to more closely match the API spec.

Reply:
...oh! That would actually make things a bit easier, I think.

Reply:
The context refactor is here I kept the existing tests intact and added some new ones for the attach/detach class methods https://github.com/open-telemetry/opentelemetry-ruby/pull/807